### ISSUE TITLE
Connect modules won't be installed

### ISSUE THREAD

**Issue by razbakov** [AOEpeople/composer-installers#1]
Connect modules are non-modman modules and throws this error:

```
ERROR: Mana_Filters does not contain a "modman" module description file.
```

**Comment by fbrnc** [comment]
Hi @razbakov,
composer-installers is only in charge of pulling in the dependencies and doesn't do anything with it. We like to use the standard modman script after running composer, but it's up to you how you want to handle "installing" the modules. Obviously modman can't handle modules downloaded from Magento Connect that don't come with a modman file.

The Hackathon Magento composer installer can do that. I don't think it doesn't belong there, but if you're looking for one tool that takes care of both steps you might want to use that one instead. Otherwise you'll have to create a modman file and inject it into your copy/fork of the module you're trying to install.

**Comment by klipach** [comment]
So if i use modules from connect20 i can not use this installer?

**Comment by fbrnc** [comment]
Yes, you can. But you have to create a modman file for that module first. Otherwise the modman script wouldn't know what to do with the module. That's exactly the part the https://github.com/magento-hackathon/magento-composer-installer would automagically do for you. But I think you should have full control over this process and use little tools that do a single thing (and that one right).
Creating a modman file for a module that doesn't already have one only takes a couple of minutes...

Also, checkout this blog post I write a while ago for a better overview of your options:
http://fbrnc.net/blog/2014/11/how-to-install-a-magento-module

**Comment by klipach** [comment]
where should i put `modman` files for modules from connect20? I have following structure in composer.json:

```
"repositories": [
    {
        "type": "composer",
        "url": "http://packages.firegento.com"
    }
],
"require": {
    "connect20/mana_filters": "13.10.17.19"
}
```

**Comment by fbrnc** [comment]
Since `connect20/mana_filters` doesn't come with a modman file (I just checked), you'd have to create your own repository and add the `modman` file there (on root level).

Then - in your `composer.json` file you need to specify that you want to pull in that module from your repo instead of `http://packages.firegento.com` (alternatively you could run your own Satis server)

```
[...]
"repositories": [
        {
            "type": "vcs",
            "url": "url pointing to your git repo"
        },
[...]
```

**Comment by klipach** [comment]
That means i should make repositories for all modules from `connect20` which are used in my project and update them manually with new versions? Not easy solution.

**Comment by fbrnc** [comment]
At this point you have following options:
- use https://github.com/magento-hackathon/magento-composer-installer instead of this
- "teach" modman how to deploy a package via `package.xml` instead of the `modman` file (or implement a package.xml to modman converter - should be pretty easy to do - and include this tool into your install script.
